lib/reverse_sentence.rb

```diff
@@ -1,6 +1,19 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) in the worst case scenario and O(n) in the best case. But since we always consider the worst case, it's O(n^2).
+# Space complexity: O(1) No matter the sentence, one array gets made, that eventually gets converted back into a string.
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  
+  return nil if my_sentence == nil
+    
+  my_sentence = my_sentence.split(/(\s)/)
```

Comment:
Since you're creating a new array of all the words this one line has O(n) space and time complexities.